Makefile

```diff
@@ -1,22 +1,110 @@
-.PHONY: build
-build:
-	go build
-
-.PHONY: install/checks/spell-and-markdown
-install/checks/spell-and-markdown:
-	yarn
-
-.PHONY: run/checks/spell-and-markdown
-run/checks/spell-and-markdown:
-	yarn check-trailing-whitespaces
-	yarn check-word-lists
-	yarn cspell --no-progress '**/**'
-	yarn markdownlint --ignore 'node_modules/' '**/*.md'
-
-.PHONY: install/checks/schema-validation
-install/checks/schema-validation:
-	cd schema-test && go mod download
-
-.PHONY: run/checks/schema-validation
-run/checks/schema-validation:
-	cd schema-test && go clean -testcache && go test .
+.DEFAULT_GOAL := help
+MAKEFLAGS += --silent --no-print-directory
+
+BIN_DIR := ./.bin
+SCRIPTS_DIR := ./internal/scripts
+
+# Print Makefile target step description for check.
+# Only print 'check' steps this way, and not dependent steps, like 'install'.
+# ${1} - step description
+define _print_step
+	printf -- '------\n%s...\n' "${1}"
+endef
+
+## Activate developer environment using devbox. Run `make install/devbox` first If you don't have devbox installed.
+activate:
+	devbox shell
+
+## Install devbox binary.
+install/devbox:
+	curl -fsSL https://get.jetpack.io/devbox | bash
+
+.PHONY: test test/go/unit
+## Run all tests.
+test: test/go/unit
+
+## Run Go unit tests.
+test/go/unit:
+	$(call _print_step,Running Go unit tests)
+	go test -race -cover ./...
+
+.PHONY: check check/vet check/lint check/gosec check/spell check/trailing check/markdown check/format
+## Run all checks.
+check: check/vet check/lint check/gosec check/spell check/trailing check/markdown check/format
+
+## Run 'go vet' on the whole project.
+check/vet:
+	$(call _print_step,Running go vet)
+	go vet ./...
+
+## Run golangci-lint all-in-one linter with configuration defined inside .golangci.yml.
+check/lint:
+	$(call _print_step,Running golangci-lint)
+	golangci-lint run
+
+## Check for security problems using gosec, which inspects the Go code by scanning the AST.
+check/gosec:
+	$(call _print_step,Running gosec)
+	gosec -exclude-dir=test -exclude-generated -quiet ./...
+
+## Check spelling, rules are defined in cspell.json.
+check/spell:
+	$(call _print_step,Verifying spelling)
+	yarn --silent cspell --no-progress '**/**'
+
+## Check for trailing whitespaces in any of the projects' files.
+check/trailing:
+	$(call _print_step,Looking for trailing whitespaces)
+	$(SCRIPTS_DIR)/check-trailing-whitespaces.bash
+
+## Check markdown files for potential issues with markdownlint.
+check/markdown:
+	$(call _print_step,Verifying Markdown files)
+	yarn --silent markdownlint '**/*.md' --ignore 'node_modules'
+
+## Check for potential vulnerabilities across all Go dependencies.
+check/vulns:
+	$(call _print_step,Running govulncheck)
+	govulncheck ./...
+
+## Verify if the files are formatted.
+## You must first commit the changes, otherwise it won't detect the diffs.
+check/format:
+	$(call _print_step,Checking if files are formatted)
+	$(SCRIPTS_DIR)/check-formatting.sh
```

Comment:
Why do we need this? We already have gofumpt to format all the files, right? So, we could just that in pre-commit hook.

Reply:
you still need that even If you have a pre-commit hook (which I personally really don't like and wouldn't want to impose upon this project)
even if you're using pre-commit hooks you still need CI to verify it (you can easily omit pre-commit hooks with `--no-verify` flag), so this script will have to stay even if we'd add pre-commit hooks